src/bin/krimedc.rs

```diff
@@ -379,6 +387,7 @@ impl TryFrom<Config> for ServerState {
                     PrincipalRecord {
                         service: false,
                         base_key,
+                        last_request_time: None,
```

Comment:
And the benefit of the builder means you don't need to have all these dangling None's here.